### PR TITLE
insert knot returns whether it was successful

### PR DESCRIPTION
**Pull request by j042**
# Overview
As the title says

## Addressed issues
*  If knot insertion fails, nothing happened

## Checklists
* [ ] Documentations are up-to-date.
* [ ] Added example(s)
* [ ] Added test(s)

### CODE REVIEW

File: cpp/splinepy/splines/helpers/scalar_type_wrapper.hpp [hunk 1]
```diff
@@ -187,12 +187,24 @@ bool ScalarTypeReduceDegree(SplineType& spline,
...
+  // see if it was successful
+  const auto& knot_vector =
+      *spline.GetParameterSpace().GetKnotVectors()[query_dim];
+  const auto n_knots_before = knot_vector.GetSize();
+
   spline.InsertKnot(Dim{query_dim}, Knot{query});
+
+  const auto n_knots_after = knot_vector.GetSize();
+
+  return (n_knots_after > n_knots_before) ? true : false;
```
Comment (jzwar) [inline cpp/splinepy/splines/helpers/scalar_type_wrapper.hpp:207]: Ehm
```suggestion
  return (n_knots_after > n_knots_before);
```

**Comment by jzwar** [comment]
We would also need to update the members, right?

**Review by jzwar — APPROVED** [review]
Thanks for changing this so quickly! LGTM

**Comment by j042** [comment]
@jzwar now the derived BSpline families should return the results

**Review by jzwar — APPROVED** [review]
Thanks for changing this so quickly, ready for merge, please bump the version before doing so!

**Pull request merged**